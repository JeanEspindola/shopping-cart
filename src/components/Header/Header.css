.header {
  margin-top: 20px;
  margin-bottom: 10px;
}

.header__title {
  font-size: 30px;
  font-weight: 500;
}

.header__toolbar {
  float: right;
}