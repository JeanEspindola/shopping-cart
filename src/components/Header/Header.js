/*
* Header Container Component.
*/
import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, ButtonToolbar, Button, Glyphicon } from 'react-bootstrap';
import './Header.css';

const Header = ({ onClearCart }) => (
  <Row className="header">
    <Col xs={6}>
      <span className="header__title">
          Your Cart
      </span>
    </Col>
    <Col xs={6}>
      <ButtonToolbar className="header__toolbar">
        <Button bsStyle="danger" onClick={onClearCart}>
          <Glyphicon glyph="trash" /> Clear Cart
        </Button>
        <Button bsStyle="primary"><Glyphicon glyph="plus" /> Add New Product</Button>
      </ButtonToolbar>
    </Col>
  </Row>
);

Header.propTypes = {
  onClearCart: PropTypes.func.isRequired,
};

export default Header;
