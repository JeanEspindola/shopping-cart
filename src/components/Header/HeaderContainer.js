/*
* Header Container Component.
*
* Connects the component with redux.
* Dispatches the cleanUpCartList action.
*
*/
import { connect } from 'react-redux';
import { cleanUpCartList } from '../../actions/cartListAction';
import Header from './Header';

const mapDispatchToProps = dispatch => ({
  onClearCart: () => dispatch(cleanUpCartList()),
});

export default connect(null, mapDispatchToProps)(Header);
