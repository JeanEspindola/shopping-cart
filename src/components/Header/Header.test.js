import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from './Header';

describe('Header', () => {
  const mockClearCart = jest.fn;

  let Component;

  beforeEach(() => {
    Component = shallow(<Header
      onClearCart={mockClearCart}
    />);
  });

  it('renders Header without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });
});
