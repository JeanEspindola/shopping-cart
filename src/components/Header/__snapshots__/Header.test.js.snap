// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header without crashing 1`] = `
<div>
  <Row
    bsClass="row"
    className="header"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      xs={6}
    >
      <span
        className="header__title"
      >
        Your Cart
      </span>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      xs={6}
    >
      <ButtonToolbar
        bsClass="btn-toolbar"
        className="header__toolbar"
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="danger"
          disabled={false}
          onClick={[Function]}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="trash"
          />
          Clear Cart
        </Button>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="plus"
          />
          Add New Product
        </Button>
      </ButtonToolbar>
    </Col>
  </Row>
  <Connect(CartItemForm)
    onClose={[Function]}
    show={false}
    title="Add Product"
  />
</div>
`;
